### DISCUSSION TITLE
SimpleSafe "not responding" - is it Oauth invalidation?

### DISCUSSION THREAD

**Discussion by brianysf** [homebridge-simplisafe3/homebridge-simplisafe3#339] [General]
I don't know if this is related to the new authorization method in 1.10.4 (no more cut and pasting mobile URL, hooray!) but the connection seems to "break" more often now. When this happens I don't see any errors in the logs but the Simplisafe status doesn't update in Homebridge and HomeKit reports "Device isn't responding."

I restart Homebridge and notice that authentication fails. I click on "Show authentication form" and reenter my credentials and all is fine after another restart.

Details:
- Simplisafe connected on WiFi, with spotty LTE connection.
- 2-factor authentication enabled

**Comment by shamoon** [comment]
The mechanism to maintain the connection hasn't really changed (maybe a tiny bit) so not sure its related. If you can enable `debug` on the plugin (and restart) it might be more helpful. The plugin should preemptively re-authenticate < 1x per hour. For me this works consistently but not so for others, I have seen. if it fails though, there should be log output.
